app/javascript/components/patient/vaccines/VaccineModal.js

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Modal, Row, Col, Form } from 'react-bootstrap';
+import Select from 'react-select';
+import moment from 'moment';
+import ReactTooltip from 'react-tooltip';
+
+import DateInput from '../../util/DateInput';
+
+class VaccineModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product_name: this.props.currentVaccineData?.product_name,
+      // If there is not already a group name, default to the first option
+      group_name: this.props.currentVaccineData?.group_name
+        ? this.props.currentVaccineData.group_name
+        : this.props.group_name_options
+        ? this.props.group_name_options[0]
+        : null,
+      administration_date: this.props.currentVaccineData?.administration_date,
+      dose_number: this.props.currentVaccineData?.dose_number,
+      notes: this.props.currentVaccineData?.notes,
+      // Sorting so that the blank option shows up at the top
+      sorted_dose_number_options: this.props.dose_number_options ? this.props.dose_number_options.sort() : [],
+      isValid: false,
+    };
+  }
+
+  componentDidMount() {
+    this.checkIfValid();
+  }
+
+  /**
+   * Data is valid and can be saved if there is both a group name and product name selected.
+   */
+  checkIfValid = () => {
+    const isValid = this.state.group_name && this.state.product_name;
+    this.setState({ isValid });
+  };
+
+  handleProductNameChange = data => {
+    this.setState({ product_name: data.value }, () => {
+      this.checkIfValid();
```

Comment:
This feels awkward; I'd propose `checkIfValid` does not change state, but instead just reports if things are valid, and set `isValid` at the same time as the first setState, rather than using the callback like this.

Reply:
what about just adding:
```
const isValid = this.state.group_name && this.state.product_name;
```
to the beginning of `render()`
and calling `!isValid` instead of `!this.state.isValid` on line 221?

Reply:
you could even just add `this.state.group_name && this.state.product_name` to determine the disabled field

Reply:
Yeah I originally was thinking the logic might involve more fields and overcomplicated it. Updated here: https://github.com/SaraAlert/SaraAlert/pull/721/commits/7b5ec1bd59c6c8c33fdb9331c5cf35933767c972#diff-182e13a82e02db14eb5222d1eb289b14639dc17acd927608ff7e8b032cfb9af8R68